dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Helper/ProviderServiceWrapper.cs

```diff
@@ -1,42 +1,88 @@
+using LazyCache;
 using Microsoft.Extensions.Options;
 
 namespace Dfc.Providerportal.FindAnApprenticeship.Helper
 {
     public class ProviderServiceWrapper : IProviderServiceWrapper
     {
+        private readonly IAppCache _cache;
         private readonly IProviderServiceSettings _settings;
 
-        public ProviderServiceWrapper(IOptions<ProviderServiceSettings> settings)
+        public ProviderServiceWrapper(IOptions<ProviderServiceSettings> settings, 
+            IAppCache cache)
         {
+            Throw.IfNull(cache, nameof(cache));
             Throw.IfNull(settings, nameof(settings));
-            
+
+            _cache = cache;
             _settings = settings.Value;
         }
+
+        /// <summary>
+        /// Mostly returns a single provider, but in some cases, we have multiple orgs with the same UKPRN. Quirky!
+        /// </summary>
+        /// <param name="UKPRN">The UKPRN to lookup</param>
+        /// <returns>A list of matching providers.</returns>
         public IEnumerable<Provider> GetProviderByUKPRN(string UKPRN)
         {
-            // Call service to get data
-            HttpClient client = new HttpClient();
-            client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", _settings.ApiKey);
-            var response = client.GetAsync($"{_settings.ApiUrl}GetProviderByPRN?PRN={UKPRN}").Result;
-            if (response.IsSuccessStatusCode)
+            try
+            {
+                return this.GetAllProviders().Where(x => x.UnitedKingdomProviderReferenceNumber == UKPRN);
+            }
+            catch (Exception e)
+            {
+                throw new ReferenceDataServiceException(UKPRN, e);
+            }
+
+        }
+
+        public IEnumerable<Provider> GetAllProviders()
+        {
+            Func<IEnumerable<Provider>> activeProvidersGetter = () => PopulateActiveProvidersCache();
+
+            try
             {
-                var json = response.Content.ReadAsStringAsync().Result;
-                if (!json.StartsWith("["))
-                    json = "[" + json + "]";
-                client.Dispose();
-                return JsonConvert.DeserializeObject<IEnumerable<Provider>>(json);
+                return _cache.GetOrAdd("ActiveProviders", activeProvidersGetter, DateTimeOffset.Now.AddHours(8));
             }
-            client.Dispose();
-            return new List<Provider>();
+            catch (HttpRequestException e)
+            {
+                // add polly retry
+                throw new ReferenceDataServiceException(e);
+            }
+            catch (Exception e)
+            {
+                throw new ReferenceDataServiceException(e);
+            }
+        }
+
+
+        private IEnumerable<Provider> PopulateActiveProvidersCache()
+        {
+            Console.WriteLine($"[{DateTime.UtcNow:G}] Cache missing or expired... Refreshing Active Providers cache");
+
+            HttpClient client = new HttpClient();
```

Comment:
Nit: re-use this or Dispose() it; we've seen perf issues with having too many of these guys around

Reply:
Apologies - this is a copy/paste of the as-is functionality... Will switch this morning.

Reply:
Rewrote the Services to take advantage of HttpClientFactory. Clients defaulted to a 2 minute lifetime